app/models/manageiq/providers/cloud_manager/provision/state_machine.rb

```diff
@@ -37,6 +37,12 @@ def start_clone_task
     phase_context[:clone_task_ref] = start_clone(phase_context[:clone_options])
     phase_context.delete(:clone_options)
     signal :poll_clone_complete
+  rescue Azure::Armrest::BadRequestException => err
+    phase_context[:exception_class]     = err.class.name
+    phase_context[:exception_message]   = err.message
+    phase_context[:exception_backtrace] = err.backtrace
+    phase_context[:error_phase]         = phase
+    signal :clone_failure_cleanup
```

Comment:
Oh sorry this should be in the azure provider, should have been more clear about that.  You can override this method there an implement the specific rescue code there.